src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -449,7 +449,22 @@ public static Type ToReflectionOnlyType(Type type)
 
         public static IEnumerable<Type> GetTypes(Assembly assembly, Func<Type, bool> whereFunc)
         {
-            return assembly.IsDynamic ? Enumerable.Empty<Type>() : assembly.DefinedTypes.Where(type => !type.GetTypeInfo().IsNestedPrivate && whereFunc(type));
+            if (assembly.IsDynamic)
+            {
+                return Enumerable.Empty<Type>();
+            }
+
+            IEnumerable<TypeInfo> definedTypes;
+            try
+            {
+                definedTypes = assembly.DefinedTypes;
+            }
+            catch (ReflectionTypeLoadException)
```

Comment:
Are there any other exceptions which we need to catch? Should we catch all exceptions?
We should log these exceptions.